api/envoy/extensions/filters/http/on_demand/v3/on_demand.proto

```diff
@@ -10,11 +12,19 @@ option java_outer_classname = "OnDemandProto";
 option java_multiple_files = true;
 option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
-// [#protodoc-title: OnDemand]
-// IP tagging :ref:`configuration overview <config_http_filters_on_demand>`.
+// [#protodoc-title: On Demand Discovery]
+// On Demand Discovery :ref:`configuration overview <config_http_filters_on_demand>`.
 // [#extension: envoy.filters.http.on_demand]
 
+// On Demand Discovery filter config.
 message OnDemand {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.http.on_demand.v2.OnDemand";
+
+  config.core.v3.ConfigSource odcds_config = 1;
```

Comment:
If the only knob here is the presence of a `ConfigSource`, then how will we configure ODCDS when we're using new-style resource names and don't need a `ConfigSource` because we're using authority-based routing?

Reply:
I think the presence of an OCDS `PerRouteConfig` would indicate we needed to treat the name as on-demand. The meaning of `self` becomes interesting as it might mean two different things for the filter-level config (received from LDS) and the `PerRouteConfig` (from RDS).